conduit/plugins/processors/filterprocessor/fields/searcher.go

```diff
@@ -18,28 +18,36 @@ type Searcher struct {
 	MethodToCall string
 }
 
+func MakeSearcher(filterTag string, expressionType expression.FilterType, expressionStr string) (*Searcher, error) {
+	t, err := LookupFieldByTag(filterTag, &transactions.SignedTxnInBlock{})
+	if err != nil {
+		return nil, err
+	}
+
+	exp, err := expression.MakeExpression(expressionType, expressionStr, t)
+	if err != nil {
+		return nil, fmt.Errorf("filter processor Init(): could not make expression with string %s for filter tag %s - %w", expressionStr, filterTag, err)
+	}
+
+	searcher, err := MakeFieldSearcher(exp, expressionType, filterTag)
+	if err != nil {
+		return nil, fmt.Errorf("filter processor Init(): error making field searcher - %w", err)
+	}
+
+	return searcher, nil
+}
+
 // This function is ONLY to be used by the filter.field function.
 // The reason being is that without validation of the tag (which is provided by
 // MakeFieldSearcher) then this can panic
-func (f Searcher) search(input transactions.SignedTxnInBlock) (bool, error) {
+func (f Searcher) search(input *transactions.SignedTxnInBlock) (bool, error) {
 
-	val, err := SignedTxnFunc(f.Tag, &input)
+	val, err := LookupFieldByTag(f.Tag, input)
 	if err != nil {
 		return false, err
 	}
 
-	e := reflect.ValueOf(val).Elem()
-
-	var toSearch interface{}
-	if f.MethodToCall != "" {
-		// If there is a function, search what is returned
-		toSearch = e.MethodByName(f.MethodToCall).Call([]reflect.Value{})[0].Interface()
-	} else {
-		// Otherwise, get the original value
-		toSearch = e.Interface()
-	}
-
-	b, err := (*f.Exp).Search(toSearch)
+	b, err := (*f.Exp).Search(val)
```

Comment:
This degree of abstraction wasn't necessary. The only "MethodToCall" is "String", the only place it's used is in the `regexExpression`. So I replaced it with `fmt.Sprintf("%s", input)` in the `Search` function. This yielded a performance improvement as well.